src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/DetectorOptions.java

```diff
@@ -69,6 +71,7 @@
      */
     public DetectorOptions setFormat(String format) {
         this.format = format;
+        pattern = null;
```

Comment:
he need to reset pattern each time smb/smth call setters that are constituents of Pattern: ignore case option and format.